frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx

```diff
@@ -5,22 +5,61 @@ import { FormikValues, useFormikContext } from 'formik';
 import { GridItem } from '@patternfly/react-core';
 import { coFetchJSON, coFetch } from '@console/internal/co-fetch';
 import { DropdownField } from '@console/shared';
-import { HelmChartMetaData, HelmRelease, HelmChart } from '../helm-types';
+import { confirmModal } from '@console/internal/components/modals/confirm-modal';
+import { HelmChartMetaData, HelmRelease, HelmChart, HelmActionType } from '../helm-types';
 import { getChartURL, getChartVersions, getChartValuesYAML } from '../helm-utils';
 
 export type HelmChartVersionDropdownProps = {
   chartVersion: string;
+  appVersion: string;
   chartName: string;
+  helmActionType: string;
 };
+type ModalCallback = () => void;
 
 const HelmChartVersionDropdown: React.FunctionComponent<HelmChartVersionDropdownProps> = ({
   chartVersion,
+  appVersion,
   chartName,
+  helmActionType,
 }) => {
-  const { setFieldValue } = useFormikContext<FormikValues>();
+  const {
+    setFieldValue,
+    values: { isYAMLDirty },
+    setFieldTouched,
+  } = useFormikContext<FormikValues>();
   const [helmChartVersions, setHelmChartVersions] = React.useState({});
   const [helmChartEntries, setHelmChartEntries] = React.useState<HelmChartMetaData[]>([]);
 
+  const warnOnChartVersionChange = (
+    onAccept: ModalCallback,
+    currentVersion: string,
+    newVersion: string,
+  ) => {
+    confirmModal({
+      title: 'Change Chart Version?',
+      message: (
+        <>
+          Are you sure you want to change the chart version from <strong>{currentVersion}</strong>{' '}
+          to <strong>{newVersion}</strong>? <br />
+          You have data entered for version <strong>{currentVersion}</strong> in the YAML editor.
+          All data entered will be lost when changed to <strong>{newVersion}</strong>
+        </>
+      ),
+      submitDanger: true,
+      btnText: 'Yes',
+      cancelText: 'No',
+      executeFn: () => {
+        onAccept();
+        return Promise.resolve();
+      },
+      cancel: () => {
```

Comment:
because, when I select an option from the dropdown the chartVersion is immediately set to the selected option and the modal opens up. Now, if I cancel the modal the chart version is not reset to the initial version, which is why I had to manually do the reset by setting the chartversion value to the initial version.